transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -534,6 +627,20 @@ boolean writeComplete0(byte op, int res, int flags, short data, int outstanding)
                 iovArray.release();
             }
             if (res >= 0) {
+                if (IoUring.isIOUringSendZCSupported()) {
+                    //a first CQE with standard res semantics and the IORING_CQE_F_MORE flag set, and
+                    if ((flags & Native.IORING_CQE_F_MORE) != 0) {
+                        lasSendZCRes = res;
+                        return true;
+                    }
+                    //a second CQE with res set to 0 and the IORING_CQE_F_NOTIF flag set.
+                    if ((flags & Native.IORING_CQE_F_NOTIF) != 0) {
+                        channelOutboundBuffer.removeBytes(lasSendZCRes);
+                        lasSendZCRes = -1;
```

Comment:
I think you should reorder as removeBytes(..) might cause a promise to be notified and so a listener to be called.
```suggestion
                        lasSendZCRes = -1;
                        channelOutboundBuffer.removeBytes(lasSendZCRes);
```